selene-vscode/src/extension.ts

```diff
@@ -165,19 +168,34 @@ export async function activate(context: vscode.ExtensionContext) {
         diagnosticsCollection.set(document.uri, diagnostics)
     }
 
+    let lastTimeout: NodeJS.Timeout
     function listenToChange() {
+        const idleDelay = vscode.workspace.getConfiguration("selene").get<number>("idleDelay") as number
         switch (vscode.workspace.getConfiguration("selene").get<RunType>("run")) {
             case RunType.OnSave:
                 return vscode.workspace.onDidSaveTextDocument(lint)
             case RunType.OnType:
                 return vscode.workspace.onDidChangeTextDocument(event => lint(event.document))
-
+            case RunType.OnNewLine:
+				return vscode.workspace.onDidChangeTextDocument(event => {
+					// Contrary to removing lines, adding new lines will leave the range at the same value hence the string comparisons
```

Comment:
I don't understand this, what if we make a new line in the middle of the script?

Reply:
`range.start.line` and `range.end.line` will stay the exact same as a new line is added. While removing lines they are different. `isSingleLine` is a `boolean` to determine if the range is on different lines which still respects what I’ve said above.
If you make a new line in the middle of the script, then the text would either be `\n` or `\r\n`